src/averageAges.js

```diff
@@ -20,6 +20,24 @@ function calculateMenAverageAge(people, century) {
   // avoid using loop and forEach
   // replace `if ()` statement with &&, || or ?:
   // without nesting
+  const mans = people.filter(person => person.sex === 'm');
+
+  if (century) {
+    const mansRightAge = mans.filter(person =>
+      Math.ceil(person.died / 100) === century);
+
+    const newArr = mansRightAge.map((person) => person.died - person.born);
+
+    const averageAge = newArr.reduce((sum, element) => sum + element, 0);
+
+    return averageAge / newArr.length;
```

Comment:
this part is the same for 2 conditions
remove it from if else block and put later in code

your condition should look like this

```
let women = people.filter(person => person.sex === 'f');

  if (withChildren) {
    women = ... women only with children;
  }

..find average ages no matter with or without children, cause you've handled it before
```